### PR TITLE
Auto-upgrade CustomScript for Splunk.

### PR DESCRIPTION
**Pull request by boumenot**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@boumenot__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (chrboum). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: splunk-on-ubuntu/splunk-2disk-resources-loop.json [hunk 1]
```diff
@@ -166,6 +166,7 @@
         "publisher": "Microsoft.OSTCExtensions",
         "type": "CustomScriptForLinux",
         "typeHandlerVersion": "1.4",
```
Comment (singhkay) [inline splunk-on-ubuntu/splunk-2disk-resources-loop.json]: Upgrade to 1.5

File: splunk-on-ubuntu/splunk-2disk-resources.json [hunk 2]
```diff
@@ -149,6 +149,7 @@
         "publisher": "Microsoft.OSTCExtensions",
         "type": "CustomScriptForLinux",
         "typeHandlerVersion": "1.4",
```
Comment (singhkay) [inline splunk-on-ubuntu/splunk-2disk-resources.json]: Upgrade to 1.5

File: splunk-on-ubuntu/splunk-8disk-resources-loop.json [hunk 3]
```diff
@@ -225,6 +225,7 @@
         "publisher": "Microsoft.OSTCExtensions",
         "type": "CustomScriptForLinux",
         "typeHandlerVersion": "1.4",
```
Comment (singhkay) [inline splunk-on-ubuntu/splunk-8disk-resources-loop.json]: Upgrade to 1.5

File: splunk-on-ubuntu/splunk-8disk-resources.json [hunk 4]
```diff
@@ -209,6 +209,7 @@
         "publisher": "Microsoft.OSTCExtensions",
         "type": "CustomScriptForLinux",
         "typeHandlerVersion": "1.4",
```
Comment (singhkay) [inline splunk-on-ubuntu/splunk-8disk-resources.json]: Upgrade to 1.5

**Comment by singhkay** [comment]
@rarsan As the template author, you good with this change?

**Comment by rarsan** [comment]
Thanks @boumenot for the change.
Doesn't autoUpgradeMinorVersion effectively remove version pinning, thereby adding risk of instability?

**Comment by boumenot** [comment]
It removes version pinning for minor versions, it does **not** do so for patch versions.  If a new patch version is released of an extension, it will be installed on the VM automatically.

The intent of this change is to opt in templates for extension testing.  We would like to test extensions against as many templates as possible.  Having a well known source of working templates that auto upgrade makes that easier to manage.

**Comment by rarsan** [comment]
Thx for explaining the rationale. LGTM.

**Comment by singhkay** [comment]
Thanks @rarsan, merged.

**Comment by boumenot** [comment]
@rarsan, I can see from recent commits that you removed the autoUpgradeMinorVersion support.  Is there a particular reason why?  I would like to understand the motivation, so we can determine the best way to address it.

**Comment by rarsan** [comment]
@boumenot , sorry for the delayed reply. I was busy with other projects.
There's no particular reason for that change. It's a side effect of using similar base code for marketplace solution. I'm about to submit a new PR where I can add this back in. Stay tuned.

**Comment by rarsan** [comment]
@boumenot please refer to #2196

**Comment by boumenot** [comment]
np.  Glad to hear it was something simple, and that it is back.  This help immensely, thank you!

**Pull request merged**